packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx

```diff
@@ -79,49 +72,113 @@ export const DeclarativeChart: React.FunctionComponent<DeclarativeChartProps> =
   DeclarativeChartProps
 >((props, forwardedRef) => {
   const { plotlySchema } = props.chartSchema;
-  const xValues = plotlySchema.data[0].x;
+  const { data, layout, selectedLegends } = plotlySchema;
+  const xValues = data[0].x;
   const isXDate = isDateArray(xValues);
   const isXNumber = isNumberArray(xValues);
   const colorMap = useColorMapping();
   const isDarkTheme = UseIsDarkTheme();
 
-  switch (plotlySchema.data[0].type) {
+  const [activeLegends, setActiveLegends] = React.useState<string[]>(selectedLegends ?? []);
+  const onActiveLegendsChange = (keys: string[]) => {
+    setActiveLegends(keys);
+    if (props.onSchemaChange) {
+      props.onSchemaChange({ plotlySchema: { data, layout, selectedLegends: keys } });
+    }
+  };
+
+  let legendProps;
+  if (activeLegends.length > 0) {
+    legendProps = {
+      canSelectMultipleLegends: false,
+      selectedLegend: activeLegends[0],
+      onChange: onActiveLegendsChange,
+    };
+  } else {
+    legendProps = {
+      canSelectMultipleLegends: false,
+      onChange: onActiveLegendsChange,
+    };
+  }
+
```

Comment:
We can set selectedLegend = ...(activeLegends.length > 0 && { selectedLegend: activeLegends[0] }) to avoid if-else.